project1-sps/script.js

```diff
@@ -1,4 +1,39 @@
-var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+var main = function (playerChoice) {
+  var computerChoice = Math.floor(Math.random() * 3) + 1;
+  if (computerChoice === 1) {
+    computerChoice = "scissors";
+  } else if (computerChoice === 2) {
+    computerChoice = "paper ";
+  } else if (computerChoice === 3) {
+    computerChoice = "stone";
+  }
+
+  if (
+    (playerChoice === "scissors" && computerChoice === "paper") ||
+    (playerChoice === "paper" && computerChoice === "stone") ||
+    (playerChoice === "stone" && computerChoice === "scissors")
+  ) {
+    return `You win! <br>
+    You played ${playerChoice} and the computer played ${computerChoice}. <br><br>
+    Type 'scissors', 'paper', or 'stone' to play another round.`;
+  } else if (playerChoice === computerChoice) {
+    return `It's a draw. <br>
+    You played You played ${playerChoice} and the computer played ${computerChoice}. <br><br>
+    Type 'scissors', 'paper', or 'stone' to play another round.`;
+  } else if (
+    (playerChoice === "scissors" && computerChoice === "stone") ||
+    (playerChoice === "paper" && computerChoice === "scissors") ||
+    (playerChoice === "stone") & (computerChoice === "paper")
+  ) {
+    return `You lose! <br>
+    You played ${playerChoice} and the computer played ${computerChoice}. <br><br>
+    Type 'scissors', 'paper', or 'stone' to play another round.`;
```

Comment:
this is ok but you are not using the full benefits of early return. recall that when we return, the rest of the function will stop running. the benefit of doing an early return is that we do not need to nest our logic in many layers of `else if` and `else` which can be unreadable and hard to follow esp if its a more complex project with many layers of nesting

Reply:
what i mean by this and how i would refactor this code:

```
if ( /* win condition */ ) {
    return `You win! <br>
    You played ${playerChoice} and the computer played ${computerChoice}. <br><br>
    Type 'scissors', 'paper', or 'stone' to play another round.`;
}

if ( /* draw condition */ ) {
    return `It's a draw. <br>
    You played You played ${playerChoice} and the computer played ${computerChoice}. <br><br>
    Type 'scissors', 'paper', or 'stone' to play another round.`;
}

return `You lose! <br>
    You played ${playerChoice} and the computer played ${computerChoice}. <br><br>
    Type 'scissors', 'paper', or 'stone' to play another round.`;
```